flexget/plugins/sites/hliang.py

```diff
@@ -0,0 +1,43 @@
+from __future__ import unicode_literals, division, absolute_import
+
+import logging
+import re
+
+from flexget import plugin
+from flexget.event import event
+from flexget.plugins.internal.urlrewriting import UrlRewritingError
+from flexget.utils.soup import get_soup
+
+log = logging.getLogger('hliang')
+
+class UrlRewriteHliang(object):
+    """Hliang urlrewriter."""
+
+    # urlrewriter API
+    def url_rewritable(self, task, entry):
+        url = entry['url']
+        if url.startswith('http://bt.hliang.com/show.php'):
+            return True
+        return False
+
+    # urlrewriter API
+    def url_rewrite(self, task, entry):
+        entry['url'] = self.parse_download_page(entry['url'], task.requests)
+
+    @plugin.internet(log)
+    def parse_download_page(self, url, requests):
+        txheaders = {'User-agent': 'Mozilla/4.0 (compatible; MSIE 5.5; Windows NT)'}
+        try:
+            page = requests.get(url, headers=txheaders)
```

Comment:
I apologize for the questions, I'm not super experienced in python programming.  I appreciate any insight you can give me.

So just to clarify a bit:

1.  Do you have an example somewhere that demonstrates what you're looking for here?  I based my code from Flexget/flexget/plugins/sites/deadfrog.py, which doesn't even put the request call in the try block (a bug, I guess).  I took a brief look at other py files in the sites folder and they are all very similar.

2.  In my experience with C++/C# programming, when a client calls an interface, it needs to know what exception types to expect if the operation fails.  UrlRewritingError seems to be this agreed exception type.  

Therefore my thought was it is not appropriate to leak exceptions solely due to implementation details out to the caller.  I could one day change my implementation and throw a completely different set of exceptions.  Catching all exception types, wrapping it in UrlRewritingError, and rethrowing allows the caller to only ever expect one type of exception.  It seems like the right thing to do.

3.  I don't quite understand the logic behind using different try blocks for different function calls.  Yes each one can fail independently, but if all I'm going to do is wrap the exception and throw UrlRewritingError, why is it better for debugging and maintainability for separate try-blocks?  We can set breakpoints to figure out which line threw exceptions.  

And is having separate try blocks really that much maintainable than a single try with all your implementation in it?  I'd almost argue it seems like the code is more readable with a single try-block.